src/Components/CardboardModal/CardboardModal.tsx

```diff
@@ -0,0 +1,129 @@
+    IStackTokens
+} from '@fluentui/react';
+import { useId } from '@fluentui/react-hooks';
+import {
+    ICardboardModalProps,
+    ICardboardModalStyleProps,
+    ICardboardModalStyles
+} from './CardboardModal.types';
+import { getStyles } from './CardboardModal.styles';
+import { useTranslation } from 'react-i18next';
+
+const stackTokens: IStackTokens = { childrenGap: 8 };
+
+const getClassNames = classNamesFunction<
+    ICardboardModalStyleProps,
+    ICardboardModalStyles
+>();
+
+const CardboardModal: React.FC<ICardboardModalProps> = (props) => {
+    const {
+        isOpen,
+        onDismiss,
+        primaryButtonProps,
+        title,
+        children,
+        destructiveButtonProps,
+        styles,
+        subTitle,
+        titleIconName
+    } = props;
+
+    // contexts
+
+    // state
+
+    // hooks
+    const { t } = useTranslation();
+    const titleId = useId('modal-title');
+
+    // callbacks
+
+    // side effects
+
+    // styles
+    const classNames = getClassNames(styles, {
+        theme: useTheme(),
+        isDestructiveFooterActionVisible: !!destructiveButtonProps
+    });
+
+    return (
+        <Modal
+            isOpen={isOpen}
+            titleAriaId={titleId}
+            onDismiss={onDismiss}
+            styles={classNames.subComponentStyles.modal}
+        >
+            <Stack tokens={stackTokens} style={{ height: '100%' }}>
+                <div className={classNames.headerContainer}>
+                    <div className={classNames.titleContainer}>
+                        {titleIconName && (
+                            <Icon
+                                iconName={titleIconName}
+                                styles={classNames.subComponentStyles.icon}
+                            />
+                        )}
+                        <h3 id={titleId} className={classNames.title}>
+                            {typeof title === 'function' ? title() : title}
+                        </h3>
+                    </div>
+                    {subTitle && (
+                        <p className={classNames.subtitle}>
+                            {typeof subTitle === 'function'
+                                ? subTitle()
+                                : subTitle}
+                        </p>
+                    )}
+                </div>
+                <div className={classNames.content}>
+                    <Stack tokens={stackTokens}>{children}</Stack>
+                </div>
+                <div className={classNames.footer}>
+                    <Stack
+                        horizontal
+                        tokens={stackTokens}
+                        styles={classNames.subComponentStyles.footerStack}
+                    >
+                        {destructiveButtonProps && (
+                            <PrimaryButton
+                                {...destructiveButtonProps}
+                                styles={classNames.subComponentStyles.destructiveButton?.()}
```

Comment:
sometimes we use "informational" link on the left corner of the footer in a modal. So should we place it in the footer? 
![image](https://user-images.githubusercontent.com/45217314/193327420-90afd175-1872-4f06-8d79-bdd25e0bc1dc.png)

Reply:
I thought about that. Sure, I can add it as an alternative to the destructive action.

Reply:
Added.